text/0000-rustflags.md

```diff
@@ -0,0 +1,289 @@
+version = "0.1.0
+```
+
+cargo CLI (Output lines have been removed for simplicity):
+```
+cargo test --rustflag=-Cinstrument-coverage
+
+Compiling foo v0.1.0 (D:\projects\foo)
+    Running `rustc --crate-name foo --edition=2021 src\lib.rs ... --crate-type lib -Cinstrument-coverage`
+    Running `rustc --crate-name foo --edition=2021 src\lib.rs --test ... -Cinstrument-coverage`
+    Running `rustc --crate-name bin1 --edition=2021 src\bin\bin1.rs ... -Cinstrument-coverage`
+    Running `rustc --crate-name example --edition=2021 examples\example.rs... --crate-type bin -Cinstrument-coverage`
+    Running `rustc --crate-name foo --edition=2021 src\main.rs --test ... -Cinstrument-coverage`
+Finished test [unoptimized + debuginfo] target(s) in 1.51s
+    Running `.../debug/deps/foo-669448d9b4043564`
+
+running 1 test
+test tests::it_works ... ok
+
+test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+
+    Running `.../target/debug/deps\bin1-88f2f72473b4679f`
+
+running 0 tests
+
+test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+
+    Running `.../target/debug/deps/foo-53f9cd70087d575a`
+
+running 0 tests
+
+test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+
+Doc-tests foo
+    Running `rustdoc --edition=2021 --crate-type lib --crate-name foo --test ...`
+
+running 0 tests
+
+test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+```
+
+In the example above, the only invocation which does not include the user specified rustc flags is the invocation of
+`rustdoc`. Since `rustdoc` flags are treated different than normal `rustflags`, the flags specifed through `--rustflag=<RUSTFLAG>`
+will not be passed to the invocation of `rustdoc`.
+
+## Build scripts
+
+The new `--rustflag=<RUSTFLAG>` feature will not be passed to the invocation of rustc for build scripts that are being compiled and run
+on the host. This is currently out of scope for this RFC since rustc flags are treated differently for build scripts depending on cargo
+configuration settings as well as the target specified.
+
+## Integration with existing RUSTFLAGS
+
+In Cargo there exists numerous ways to specify which Rust compiler flags should be set when compiling a crate and its dependencies.
+
+1. `CARGO_ENCODED_RUSTFLAGS` environment variable
+2. `RUSTFLAGS` environment variable
+3. `target.*.rustflags` from the config (.cargo/config)
+4. `target.cfg(..).rustflags` from the config (.cargo/config)
+5. `host.*.rustflags` from the config (.cargo/config) if compiling a host artifact or without `--target`
+6. `build.rustflags` from the config (.cargo/config)
+
+The `--rustflag` values would override the set of rustflags calculated from the options listed above only for the current crate or the set of crates in the workspace. All upstream and transitive dependencies will still use the rustflags calculated from the environment
```

Comment:
IMO combining would be better than overriding. You should at least discuss this alternative.

Reply:
I understand, I will update the RFC to mention the default behavior as combining and add the option to override in the alternatives section. Thanks!